nxcomp/src/ChannelEndPoint.cpp

```diff
@@ -230,7 +228,9 @@ ChannelEndPoint::getUnixPath(char **unixPath) const {
       return false;
   }
 
-  *unixPath = strdup(path);
+  // Only return value wanted
+  if ( unixPath )
+    *unixPath = strdup(path);
```

Comment:
unixPath will never be NULL, so this if-clause is not required. (Remember, you added a return false further above).

Reply:
I didn't add a "return false", I removed it. The fix is that getUnixPath has to return the right boolean even when passed a NULL.

The call tree starts here:

    ChannelEndPoint::validateSpec() {
       isTCP_ = getTCPHostAndPort();
       isUnix_ = getUnixPath();


getUnixPath is declared:   

    bool getUnixPath(char **path = NULL) const;

So, unixPath is a pointer to a pointer, which will be NULL when called from validateSpec(). validateSpec() doesn't actually want the path, it only wants the true/false return value from the function.

What was broken is that the function always returned false when not given a buffer, even if it actually was an unix path.